pkg/support/runsupportbundle.go

```diff
@@ -1,172 +1,133 @@
 
 	"github.com/aws/eks-anywhere/pkg/api/v1alpha1"
 	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/filewriter"
 	"github.com/aws/eks-anywhere/pkg/logger"
 )
 
-const resultsSeparator = "\n------------\n"
-
-func ParseBundleFromDoc(clusterSpec *cluster.Spec, bundleConfig string) (*EksaDiagnosticBundle, error) {
-	af := NewAnalyzerFactory()
-	cf := NewCollectorFactory()
-	if bundleConfig == "" {
-		// user did not provide any bundle-config to the support-bundle command, generate one using the default collectors & analyzers
-		return NewBundleConfig(clusterSpec, af, cf), nil
-	}
+const (
+	troulbeshootApiVersion    = "troubleshoot.sh/v1beta2"
+	generatedBundleNameFormat = "%s-%s-bundle.yaml"
+)
 
-	// parse bundle-config provided by the user
-	collectorContent, err := supportbundle.LoadSupportBundleSpec(bundleConfig)
-	if err != nil {
-		return nil, err
-	}
-	bundle, err := supportbundle.ParseSupportBundleFromDoc(collectorContent)
-	if err != nil {
-		return nil, err
-	}
-	return NewCustomBundleConfig(bundle, af, cf), nil
-}
-
-func ParseTimeOptions(since string, sinceTime string) (*time.Time, error) {
-	var sinceTimeValue time.Time
-	var err error
-	if sinceTime == "" && since == "" {
-		return &sinceTimeValue, nil
-	} else if sinceTime != "" && since != "" {
-		return nil, fmt.Errorf("at most one of `sinceTime` or `since` could be specified")
-	} else if sinceTime != "" {
-		sinceTimeValue, err = time.Parse(time.RFC3339, sinceTime)
-		if err != nil {
-			return nil, fmt.Errorf("unable to parse --since-time option: %v", err)
-		}
-	} else if since != "" {
-		duration, err := time.ParseDuration(since)
-		if err != nil {
-			return nil, fmt.Errorf("unable to parse --since option: %v", err)
-		}
-		now := time.Now()
-		sinceTimeValue = now.Add(0 - duration)
-	}
-	return &sinceTimeValue, nil
+type EksaDiagnosticBundleOpts struct {
+	AnalyzerFactory  AnalyzerFactory
+	BundlePath       string
+	CollectorFactory CollectorFactory
+	Client           BundleClient
+	ClusterSpec      *cluster.Spec
+	Kubeconfig       string
+	Writer           filewriter.FileWriter
 }
 
 type EksaDiagnosticBundle struct {
-	bundle           *v1beta2.SupportBundle
+	bundle           *supportBundle
+	bundlePath       string
+	clusterSpec      *cluster.Spec
 	analyzerFactory  AnalyzerFactory
 	collectorFactory CollectorFactory
+	client           BundleClient
+	kubeconfig       string
+	writer           filewriter.FileWriter
 }
 
-func NewCustomBundleConfig(customBundle *v1beta2.SupportBundle, af AnalyzerFactory, cf CollectorFactory) *EksaDiagnosticBundle {
-	return &EksaDiagnosticBundle{
-		bundle:           customBundle,
-		analyzerFactory:  af,
-		collectorFactory: cf,
+func NewDiagnosticBundle(opts EksaDiagnosticBundleOpts) (*EksaDiagnosticBundle, error) {
+	if opts.BundlePath == "" && opts.ClusterSpec != nil {
+		// user did not provide any bundle-config to the support-bundle command, generate one using the default collectors & analyzers
```

Comment:
nit: we should try to not make references to commands since this is not in the cmd package. It's an independent package so it could be potentially called from somewhere else

Reply:
removed